docs/chart-concepts/templating.md

```diff
@@ -0,0 +1,116 @@
+```js
+// Celsius to Fahrenheit conversion, where point.y is degrees Celsius
+format: '{add (multiply point.y (divide 9 5)) 32}℉'
+```
+
+When doing math like this, we sometimes end up with too many decimals or other results that we want to run through number or date formatting. In that case, we use a subexpression then apply the formatting as explained above.
+
+```js
+// Celsius to Fahrenheit conversion, where point.y is degrees Celsius.
+// Format the result with 1 decimal place.
+format: '{(add (multiply point.y (divide 9 5)) 32):.1f}℉'
+```
+
+[View live demo](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/plotoptions/series-datalabels-format-subexpression).
+
+Subexpressions can also be used inside conditions to asses whether a statement is truthy.
+```js
+// A subexpression inside a condition. Decide plural form.
+format: 'The series exists of {points.length} ' +
+    '{#if (eq 1 points.length)}point{else}points{/if}.'
+```
+
+## Debugging
+One of the downsides of working with string formats versus formatter callbacks is the ability to log and debug inside callback. To alleviate that, you can define custom helpers allowing you to inspect the context.
+
+```js
+// Custom helper to log the context
+Highcharts.Templating.helpers.log = function () {
+    console.log(arguments[0].ctx);
+};
+
+// Usage
+format: '{log}'
+```
+[View live demo](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/members/format-log).
+
+## Built-in helpers
+* **add**. Add two numbers. For example `{add index 1}` where `index` is a zero-based index from the context. [Demo](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/plotoptions/series-datalabels-format-subexpression).
+* **divide**. Divide the first number with the second. For example `{divide 10 2}` prints 5. Division by zero returns an empty string. [Demo](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/plotoptions/series-datalabels-format-subexpression).
+* **eq**. Returns `true` for loose equality (JavaScript `==`) between the first and second argument. Can be used either as a block helper, `{#eq index 0}First item{/eq}`, or in a subexpression `{#if (eq index 0)}First item{/if}`.
+* **#each**. Iterate over an array of items. The context of each child is given as `{this}` in the block body. Additional variables in the block body are `@index`, `@first` and `@last`. Example `{#each points}{@index}) {name}, {#if @last}and {/if}`. [Demo](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/highcharts/tooltip/format-shared).
+* **ge**. Greater than or equal, JavaScript `>=`. Doubles as block helper and subexpression.
+* **gt**. Greater than, JavaScript `>`. Doubles as block helper and subexpression.
+* **#if**. Conditional block helper. `{#if point.isNull}The point is null{else}The value is {point.y}{/if}`.
```

Comment:
Are multiple conditions supported? Eg `{#if point.isNull || (point.y eq 0)}Zero{else}{point.y}{/if}`

Reply:
Currently not. For that we would require custom helpers for `and` and `or`. Your example would be:

```js
Highcharts.Templating.helpers.and = (a, b) => a && b;
```
`{#if (or point.isNull (eq point.y 0))}Zero{else}{point.y}{/if}`

I didn't include that in the built-in helpers because to my surprise, `#if` was hardly needed in the demos. I think it is better to add them later if requested.

Reply:
Okay, thanks!

Just the background: the idea came from `axis-setcompare`. I don't see any other (stock) demo needing this.